tests/unit/test_coordinate_utilities.py

```diff
@@ -0,0 +1,431 @@
+            )
+
+        with self.subTest(
+            'Retrieves expected override dimensions for the latitude variable'
+        ):
+            self.assertEqual(
+                get_override_projected_dimensions(self.varinfo, self.latitude),
+                expected_override_dimensions_AM,
+            )
+
+        with self.subTest(
+            'Retrieves expected override dimensions science variable with a different grid'
+        ):
+            self.assertEqual(
+                get_override_projected_dimensions(
+                    self.varinfo, '/Soil_Moisture_Retrieval_Data_PM/surface_flag_pm'
+                ),
+                expected_override_dimensions_PM,
+            )
+
+    def test_get_row_col_sizes_from_coordinate_datasets(self):
+        """Ensure that the correct row and column sizes are
+        returned for the requested coordinates
+        """
+        expected_row_col_sizes = (5, 10)
+        lat_1d_array = np.array([1, 2, 3, 4])
+        lon_1d_array = np.array([6, 7, 8, 9])
+        lat_irregular_array = np.array([[1, 2, 3], [3, 4, 5]])
+        lon_irregular_array = np.array([[6, 7], [8, 9], [10, 11]])
+        with self.subTest('Retrieves the expected row col sizes from the coordinates'):
+            self.assertEqual(
+                get_row_col_sizes_from_coordinate_datasets(self.lat_arr, self.lon_arr),
+                expected_row_col_sizes,
+            )
+        with self.subTest('Retrieves the expected row col sizes for the 1d array'):
+            self.assertEqual(
+                get_row_col_sizes_from_coordinate_datasets(lat_1d_array, lon_1d_array),
+                (4, 4),
+            )
+
+        with self.subTest(
+            'Raises an exception when the lat and lon array shapes do not match'
+        ):
+            with self.assertRaises(IrregularCoordinateVariables) as context:
+                get_row_col_sizes_from_coordinate_datasets(
+                    lat_irregular_array, lon_irregular_array
+                )
+            self.assertEqual(
+                context.exception.message,
+                f'Longitude coordinate shape: "{lon_irregular_array.shape}"'
+                f'does not match the latitude coordinate shape: "{lat_irregular_array.shape}"',
+            )
+
+    def test_get_valid_indices(self):
+        """Ensure that latitude and longitude values are correctly identified as
+        ascending or descending.
+
+        """
+        expected_valid_indices_lat_arr = np.array([0, 1, 2, 3, 4])
+        expected_valid_indices_lon_arr = np.array([0, 1, 2, 6, 7, 8, 9])
+
+        fill_array = np.array([-9999.0, -9999.0, -9999.0, -9999.0])
+        with self.subTest('valid indices with no fill values configured'):
+            valid_indices_lat_arr = get_valid_indices(
+                self.lat_arr[:, -1], None, 'latitude'
+            )
+            self.assertTrue(
+                np.array_equal(valid_indices_lat_arr, expected_valid_indices_lat_arr)
+            )
```

Comment:
`numpy` has some syntactic sugar for this:

Behold: [numpy.testing.assert_array_equal](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_equal.html)!

Reply:
updated - 36a3c4034215d85f5fccf660dc3f213d28873d3d